Makefile

```diff
@@ -1,172 +1,383 @@
+	-e GOCACHE="/tmp" \
+	$(DOCKER_IMAGE_GO) \
+	/bin/sh -c \
+	"cd /app && \
+	go test ${GO_TEST_OPTS} ./... " || ${FAIL}
+	@$(OK) testing
+
+.PHONY: go-mod-check
+go-mod-check: ## to check go mod files consistency
+	@$(INFO) Checking go mod files consistency...
+	$(AT)$(GO) mod tidy
+	$(AT)git --no-pager diff --exit-code go.mod go.sum || \
+	(${WARN} Please run "go mod tidy" and commit the changes in go.mod and go.sum. && ${FAIL} ; exit 128 )
+	@$(OK) Checking go mod files consistency
+
+.PHONY: go-update-dependencies
+go-update-dependencies: ## to update go dependencies (vendor)
+	@$(INFO) updating go dependencies...
+	$(AT)$(GO) get -u ./... && \
+	$(AT)$(GO) mod vendor && \
+	$(AT)$(GO) mod tidy || ${FAIL}
+	@$(OK) updating go dependencies
+
+.PHONY: go-lint
+go-lint: ## to lint go code
+	@$(INFO) App linting...
+	$(AT)GOCACHE="/tmp" $(DOCKER) run ${DOCKER_OPTS} \
+	-v $(PWD):/app -w /app \
+	-e GOCACHE="/tmp" \
+	-e GOLANGCI_LINT_CACHE="/tmp" \
+	${DOCKER_IMAGE_GOLINT} \
+	golangci-lint run ./... || ${FAIL}
+	@$(OK) App linting
+
+.PHONY: go-fmt
+go-fmt: ## to perform formatting
+	@$(INFO) App code formatting...
+	$(AT)$(GO) fmt ./... || ${FAIL}
+	@$(OK) App code formatting...
+
+.PHONY: go-doc
+go-doc: ## to generate documentation
+	@$(INFO) Generating Documentation...
+	$(AT)$(GO) run ./scripts/env_config.go ./docs/env_config.md || ${FAIL}
+	@$(OK) Generating Documentation
+
+.PHONY: github-release
+github-release: ## to publish a release and relevant artifacts to GitHub
+	@$(INFO) Generating github-release http://github.com/$(GITHUB_ORG)/$(GITHUB_REPO)/releases/tag/$(APP_VERSION) ...
+ifeq ($(shell echo $(APP_VERSION) | egrep '^v([0-9]+\.){0,2}(\*|[0-9]+)'),)
+	$(error "We only support releases from semver tags")
+else
+	$(AT)$(DOCKER) run \
+	-v $(PWD):/app -w /app \
+	-e GITHUB_TOKEN=${GITHUB_TOKEN} \
+	$(DOCKER_IMAGE_GH_CLI) \
+	/bin/sh -c \
+	"git config --global --add safe.directory /app && cd /app && \
+	gh release create $(APP_VERSION) --generate-notes $(GO_OUT_BIN_DIR)/*" || ${FAIL}
+endif
+	@$(OK) Generating github-release http://github.com/$(GITHUB_ORG)/$(GITHUB_REPO)/releases/tag/$(APP_VERSION) ...
 
-.PHONY: manifest
-manifest: push
-	docker manifest create --amend $(IMAGE):$(TAG) $(shell echo $(ARCHS) | sed -e "s~[^ ]*~$(IMAGE)\-&:$(TAG)~g")
-	@for platform in $(ARCHS); do docker manifest annotate --arch "$${platform}" ${IMAGE}:${TAG} ${IMAGE}-$${platform}:${TAG}; done
-	docker manifest push --purge $(IMAGE):$(TAG)
-
-test:
-	$(GO) test $(GOFLAGS) -v -timeout=180s ./...
-
-clean:
-	@echo Cleaning
-	rm -Rf $(DIST_ROOT)
-	go clean $(GOFLAGS) -i ./...
-
-run:
-	@echo Starting go web server
-	$(GO) run $(GOFLAGS) -ldflags '$(LDFLAGS)' main.go
-
-build-swagger:
-	npm run validate
-	npm run build
-
-serve-swagger:
-	npm run validate
-	npm run serve
+.PHONY: clean
+clean: ## to clean-up
+	@$(INFO) cleaning /${GO_OUT_BIN_DIR} folder...
+	$(AT)rm -rf ${GO_OUT_BIN_DIR} || ${FAIL}
+	@$(OK) cleaning /${GO_OUT_BIN_DIR} folder
```

Comment:
So much goodies in this Makefile!

Reply:
Yes, Lot's of love has gone into this Makefile. :)